### PR TITLE
Add environment variables dumping filter for native logger

### LINKED ISSUE #2871: [Debug logger] Native logger dumps secrets
### What

Currently native logger dumps all environment variables. Some of them could contain secrets or credentials, etc.

### Proposal

Filter environment variables that are dumped, if they contain:
* `API`
* `TOKEN`
* `SECRET`
* `KEY`
* `PASSWORD` / `PASS` / `PWD`
* `HEADER`
* `CREDENTIALS`
* (add missing here)

### PR DESCRIPTION
**Pull request by RassK**
## Why

Fixes #2871

## What

Does not dump environment variables to native log if specific keywords are detected (list in #2871).

## Tests

locally tested:
- [x] Windows
- [x] Linux

## Checklist

- [ ] `CHANGELOG.md` is updated.
- [ ] Documentation is updated.
- [ ] New features are covered by tests.

### CODE REVIEW

File: src/OpenTelemetry.AutoInstrumentation.Native/cor_profiler.cpp [hunk 1]
```diff
@@ -66,9 +71,23 @@ HRESULT STDMETHODCALLTYPE CorProfiler::Initialize(IUnknown* cor_profiler_info_un
         const auto env_variables = GetEnvironmentVariables(env_vars_prefixes_to_display);
         Logger::Debug("Environment variables:");
 
+#ifdef _WIN32
+        const std::regex secrets_regex("(?:^|_)(API|TOKEN|SECRET|KEY|PASSWORD|PASS|PWD|HEADER|CREDENTIALS)(?:_|$)",
```
[the remark this answers is not in the document]
Comment (RassK) [reply src/OpenTelemetry.AutoInstrumentation.Native/cor_profiler.cpp]: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/2872/commits/e28d99f412310f6c3059c5d57fd39ccbf29ed335

**Comment by RassK** [comment]
> Right now the code just outputs the env vars that are to be hidden/masked, right?

Yes, they still remain in the console output. All written logs should be clean.

**Comment by RassK** [comment]
> @RassK can you change the code to show that env var is present but not its contents?

@pjanotti do you mean for both? (console output & disk log)
I think this is actually better UX. Showing something like `OTEL_SECRET_VAR=<hidden>`

**Comment by nrcventura** [comment]
> I think this is actually better UX. Showing something like `OTEL_SECRET_VAR=<hidden>`

I like this approach, because it let's you know the setting is still there but we redacted the value.

**Comment by pjanotti** [comment]
> @pjanotti do you mean for both? (console output & disk log)

I mean for the disk log, I am a bit confused about "console output"? Are we dumping those to the stdout? I don't recall seeing that...

**Comment by RassK** [comment]
> > @pjanotti do you mean for both? (console output & disk log)
> 
> I mean for the disk log, I am a bit confused about "console output"? Are we dumping those to the stdout? I don't recall seeing that...

Ah sorry... it was the test app's code that's dumping to console.

**Review by pjanotti — APPROVED** [review]
Nice, could you attach a sample output to the description?

A follow-up issue to add a test is also good.

**Review by nrcventura — APPROVED** [review]

**Pull request merged**